spec/weak-types.html

```diff
@@ -0,0 +1,348 @@
+        </li>
+      </ul>
+
+      <p>Neither of these actions (ClearKeptObjects or CleanupFinalizationRegistry) may interrupt synchronous ECMAScript execution. Because hosts may assemble longer, synchronous ECMAScript execution runs, this specification defers the scheduling of ClearKeptObjects and CleanupFinalizationRegistry to the host environment.</p>
+
+      <p>Some ECMAScript implementations include garbage collector implementations which run in the background, including when ECMAScript is idle. Letting the host environment schedule CleanupFinalizationRegistry allows it to resume ECMAScript execution in order to run finalizer work, which may free up held values, reducing overall memory usage.</p>
+    </emu-clause>
+
+    <emu-clause id="sec-unforgeable-values">
+      <h1><ins>Unforgeable values</ins></h1>
+
+      <p>
+        <ins>A value _V_ is <dfn>unforgeable</dfn> if either one of the following conditions is true:</ins>
+        <ul>
+          <li><ins>it is an Object;</ins></li>
+          <li><ins>it is a Box whose [[Value]] internal slot is an unforgeable value;</ins></li>
+          <li><ins>it is a Tuple whose [[Sequence]] internal slot contains at least one unforgeable value;</ins></li>
+          <li><ins>it is a Record whose [[Fields]] internal slot contains at least one { [[Key]], [[Value]] } pair such that [[Value]] is an unforgeable value.</ins></li>
+        </ul>
+      </p>
+
+      <emu-note>
+        <p><ins>Symbols created by the `Symbol()` constructor are technically unforgeable: you cannot recreate one with the same identity. However, they are not covered by this definition and they are not allowed as `WeakMap` keys, `WeakSet` values, or as `WeakRef` and `FinalizationRegistry` targets.</ins></p>
+      </emu-note>
+
+      <p>
+        <ins>An unforgeable value _V_ contains an object _o_ if either one of the following conditions is true:</ins>
+        <ul>
+          <li><ins>_V_ is _o_;</ins></li>
+          <li><ins>_V_ is a Box whose [[Value]] internal slot contains _o_;</ins></li>
+          <li><ins>_V_ is a Tuple whose [[Sequence]] internal slot contains at least one unforgeable value that contains _o_;</ins></li>
+          <li><ins>_V_ is a Record whose [[Fields]] internal slot contains at least one { [[Key]], [[Value]] } pair such that [[Value]] is an unforgeable value that contains _o_.</ins></li>
+        </ul>
+      </p>
+    </emu-clause>
+
+    <emu-clause id="sec-liveness">
+      <h1>Liveness</h1>
+
+      <p>For some set of objects _S_, a <dfn>hypothetical WeakRef-oblivious</dfn> execution with respect to _S_ is an execution whereby the abstract operation WeakRefDeref of a WeakRef whose referent is <ins>an unforgeable value that <emu-xref href="#sec-unforgeable-values">contains</emu-xref></ins> an element of _S_ always returns *undefined*.</p>
```

Comment:
> whose referent is an unforgeable value that contains an element of S

In this case the referent would itself have to be in S, so `an unforgeable value that contains` should not have been added.
```suggestion
      <p>For some set of <ins>unforgeable values</ins><del>objects</del> _S_, a <dfn>hypothetical WeakRef-oblivious</dfn> execution with respect to _S_ is an execution whereby the abstract operation WeakRefDeref of a WeakRef whose referent is an element of _S_ always returns *undefined*.</p>
```